### ISSUE TITLE
dialog: Create myLeoSettings.leo crashes Leo

### ISSUE THREAD

**Issue by LewisNeal** [leo-editor/leo-editor#3335]
See PR #3831.

With leoSettings.leo open go to menu
Settings> Open myLeoSettings.leo 
opens a dialog: Create myLeoSettings.leo in C:/Users/lewisneal/.leo?
Press yes and Leo crashes:
<details>

```python
Leo 6.7.8-devel, ekr-3335-myLeoSettings.leo branch, build aa4798f45f
2024-03-05 06:53:19 -0600
Python 3.12.0, PyQt version 6.6.2
Windows 11 AMD64 (build 10.0.22631) SP0
wrote recent file: C:/Users/Dev/.leo/.leoRecentFiles.txt
Traceback (most recent call last):

  File "C:\Repos\leo-editor\leo\core\leoCommands.py", line 2722, in doCommand
    return_value = command_func(event)
                   ^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoGlobals.py", line 248, in commander_command_wrapper
    method(event=event)
  File "C:\Repos\leo-editor\leo\commands\commanderHelpCommands.py", line 199, in openMyLeoSettings
    return createMyLeoSettings(c)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\commands\commanderHelpCommands.py", line 229, in createMyLeoSettings
    c2 = g.openWithFileName(fileName, old_c=c)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoGlobals.py", line 3241, in openWithFileName
    return g.app.loadManager.openWithFileName(fileName, gui, old_c)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoApp.py", line 3007, in openWithFileName
    return lm.openEmptyLeoFile(file_name, gui, old_c)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoApp.py", line 3162, in openEmptyLeoFile
    c = g.app.newCommander(
        ^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoApp.py", line 1511, in newCommander
    c = leoCommands.Commands(fileName,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoCommands.py", line 154, in __init__
    c.finishCreate()  # Slightly slow.
    ^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoCommands.py", line 441, in finishCreate
    c.abbrevCommands.finishCreate()
  File "C:\Repos\leo-editor\leo\commands\abbrevCommands.py", line 60, in finishCreate
    self.reload_settings()
  File "C:\Repos\leo-editor\leo\commands\abbrevCommands.py", line 67, in reload_settings
    self.init_tree_abbrev()
  File "C:\Repos\leo-editor\leo\commands\abbrevCommands.py", line 206, in init_tree_abbrev
    self.init_tree_abbrev_helper(d, tree_s)
  File "C:\Repos\leo-editor\leo\commands\abbrevCommands.py", line 214, in init_tree_abbrev_helper
    hidden_root = c.fileCommands.getPosFromClipboard(tree_s)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoFileCommands.py", line 1209, in getPosFromClipboard
    v = self.getVnodeFromClipboard(s)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoFileCommands.py", line 1221, in getVnodeFromClipboard
    v = FastRead(c, {}).readFileFromClipboard(s_bytes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoFileCommands.py", line 139, in readFileFromClipboard
    hidden_v, g_element = self.readWithElementTree(path=None, s_or_b=s_or_b)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoFileCommands.py", line 181, in readWithElementTree
    self.handleBits()
  File "C:\Repos\leo-editor\leo\core\leoFileCommands.py", line 187, in handleBits
    expanded = c.db.get('expanded')
               ^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoCache.py", line 47, in get
    value = self.db.get(f"{self.c.mFileName}:::{key}")
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoCache.py", line 285, in get
    if not self.has_key(key):  # noqa
           ^^^^^^^^^^^^^^^^^
  File "C:\Repos\leo-editor\leo\core\leoCache.py", line 295, in has_key
    for _row in self.conn.execute(sql, (key,)):
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.ProgrammingError: Cannot operate on a closed database.
```

</details>

**EKR Notes**

- This crash happens only if `myLeoSettings.leo` does not already exist.
- `leoCache.py` has changed since the original report.
  I have updated the crash report accordingly.

**Comment by LewisNeal** [comment]
Please note this crash occurs if leoSettings.leo is the only file open in the session.
The crash does not occur if another Leo file is also open.

**Comment by edreamleo** [comment]
@LewisNeal Thanks for your report. I'll fix it for Leo 6.7.8.

**Comment by edreamleo** [comment]
Closed via PR #3831.
